### PR TITLE
task-WEBUI-109-MoveRetentionUI

### PR DESCRIPTION
**Pull request by nmpcunha**

### CODE REVIEW

**Review by doulba — APPROVED** [review]

**Review by andreacornaglia — APPROVED** [review]

File: nuxeo-retention-web/polymer.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "sources": [
+    "elements/**/*.js"
+  ],
+  "lint": {
+    "rules": ["polymer-3"],
+    "ignoreWarnings": []
+  },
+  "npm": true
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/polymer.json:10]: new line needed at the end of the file

File: nuxeo-retention-web/i18n/messages-ar.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "The retention period is calculated based on the retention duration starting according to a document's date field ",
+  "retention.rule.attachButton.label.heading": "Attach retention rule",
+  "retention.rule.attachButton.label.select": "Select the rule to be attached",
+  "retention.rule.attachButton.label.placeholder": "Search rule name ...",
+  "retention.rule.attachButton.label": "Attach retention rule",
+  "retention.rule.attachButton.attached": "Retention rule attached",
+  "retention.rule.attachButton.bulk.error": "An error occurred while attaching retention rule: {0}",
+  "retention.rule.attachButton.bulk.label": "Bulk attach retention rule",
+  "retention.rule.attachButton.bulk.poll": "Attaching retention rule"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-ar.json:87]: new line needed at the end of this file

File: nuxeo-retention-web/i18n/messages-de.json [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "Die Aufbewahrungszeit wird anhand der Dauer der Aufbewahrung berechnet, die nach dem Datumsfeld eines Dokuments beginnt ",
+  "retention.rule.attachButton.label.heading": "Aufbewahrungsregel anhängen ",
+  "retention.rule.attachButton.label.select": "Wähle die Regel, die hinzugefügt werden soll",
+  "retention.rule.attachButton.label.placeholder": "Suche Regelname ...",
+  "retention.rule.attachButton.label": "Aufbewahrungsregel anhängen ",
+  "retention.rule.attachButton.attached": "Aufbewahrungsregel beigefügt",
+  "retention.rule.attachButton.bulk.error": "Beim Anhängen der Aufbewahrungsregel ist ein Fehler aufgetreten: {0}",
+  "retention.rule.attachButton.bulk.label": "Aufbewahrungsregel anhängen",
+  "retention.rule.attachButton.bulk.poll": "Beifügen einer Aufbewahrungsregel"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-de.json:87]: same thing than above about a new line at the end of this file

File: nuxeo-retention-web/i18n/messages-es-ES.json [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "El período de retención se calcula en base a la duración de retención a partir del campo de fecha de un documento ",
+  "retention.rule.attachButton.label.heading": "Adjuntar regla de retención",
+  "retention.rule.attachButton.label.select": "Seleccione la regla a adjuntar",
+  "retention.rule.attachButton.label.placeholder": "Buscar nombre de regla ...",
+  "retention.rule.attachButton.label": "Adjuntar regla de retención",
+  "retention.rule.attachButton.attached": "Regla de retención adjunta",
+  "retention.rule.attachButton.bulk.error": "Se ha producido un error al adjuntar la regla de retención: {0}",
+  "retention.rule.attachButton.bulk.label": "Adjuntar en masa regla de retención",
+  "retention.rule.attachButton.bulk.poll": "Adjuntar regla de retención"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-es-ES.json:87]: same thing about new line

File: nuxeo-retention-web/i18n/messages-eu.json [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "Atxikitze-aldia kalkulatzeko atxikitzearen iraupena eta dokumentuaren data hartzen dira oinarritzat",
+  "retention.rule.attachButton.label.heading": "Erantsi atxikitze-araua",
+  "retention.rule.attachButton.label.select": "Hautatu erantsi beharreko araua",
+  "retention.rule.attachButton.label.placeholder": "Bilatu arauaren izena...",
+  "retention.rule.attachButton.label": "Erantsi atxikitze-araua",
+  "retention.rule.attachButton.attached": "Atxikitze-araua erantsia",
+  "retention.rule.attachButton.bulk.error": "Errore bat gertatu da atxikitze-araua eranstean: {0}",
+  "retention.rule.attachButton.bulk.label": "Erantsi atxikitze-arauak multzoan",
+  "retention.rule.attachButton.bulk.poll": "Atxikitze-araua eransten"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-eu.json:87]: same thing about new line

File: nuxeo-retention-web/i18n/messages-fr.json [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "La période de rétention est calculée avec pour date de départ un champ date du document",
+  "retention.rule.attachButton.label.heading": "Attacher la règle de rétention",
+  "retention.rule.attachButton.label.select": "Sélectionnez la règle à attacher",
+  "retention.rule.attachButton.label.placeholder": "Rechercher un nom de règle...",
+  "retention.rule.attachButton.label": "Attacher la règle de rétention",
+  "retention.rule.attachButton.attached": "Règle de rétention attachée",
+  "retention.rule.attachButton.bulk.error": "Une erreur est survenue lors de l'attachement de la règle de rétention : {0}",
+  "retention.rule.attachButton.bulk.label": "Appliquer la règle de rétention à une liste de documents",
+  "retention.rule.attachButton.bulk.poll": "Attacher la règle de rétention"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-fr.json:87]: same thing about new line

File: nuxeo-retention-web/i18n/messages-he.json [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "The retention period is calculated based on the retention duration starting according to a document's date field ",
+  "retention.rule.attachButton.label.heading": "Attach retention rule",
+  "retention.rule.attachButton.label.select": "Select the rule to be attached",
+  "retention.rule.attachButton.label.placeholder": "Search rule name ...",
+  "retention.rule.attachButton.label": "Attach retention rule",
+  "retention.rule.attachButton.attached": "Retention rule attached",
+  "retention.rule.attachButton.bulk.error": "An error occurred while attaching retention rule: {0}",
+  "retention.rule.attachButton.bulk.label": "Bulk attach retention rule",
+  "retention.rule.attachButton.bulk.poll": "Attaching retention rule"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-he.json:87]: same thing about new line

File: nuxeo-retention-web/i18n/messages-id.json [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  "retention.rule.attachButton.label.heading": "Attach retention rule",
+  "retention.rule.attachButton.label.select": "Select the rule to be attached",
+  "retention.rule.attachButton.label.placeholder": "Search rule name ...",
+  "retention.rule.attachButton.label": "Attach retention rule",
+  "retention.rule.attachButton.attached": "Retention rule attached",
+  "retention.rule.attachButton.bulk.error": "An error occurred while attaching retention rule: {0}",
+  "retention.rule.attachButton.bulk.label": "Bulk attach retention rule",
+  "retention.rule.attachButton.bulk.poll": "Attaching retention rule",
+  "searchAnalytics.callsPerHour.heading": "Panggilan per Jam"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-id.json:88]: same thing about new line

File: nuxeo-retention-web/i18n/messages-it.json [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "Il periodo di ritenzione è calcolato in base alla durata di ritenzione a partire dal campo data del documento ",
+  "retention.rule.attachButton.label.heading": "Allega regola di ritenzione",
+  "retention.rule.attachButton.label.select": "Seleziona la regola da allegare",
+  "retention.rule.attachButton.label.placeholder": "Ricerca nome regola...",
+  "retention.rule.attachButton.label": "Allega regola di ritenzione",
+  "retention.rule.attachButton.attached": "Regola di ritenzione allegata",
+  "retention.rule.attachButton.bulk.error": "Si è verificato un errore durante l'associazione della regola di ritenzione: {0}",
+  "retention.rule.attachButton.bulk.label": "Regola di ritenzione allegata in blocco",
+  "retention.rule.attachButton.bulk.poll": "Allega regola di ritenzione"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-it.json:87]: same thing about new line

File: nuxeo-retention-web/i18n/messages-ja.json [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "保持期間は、文書の日付フィールドに従って開始される保持期間に基づいて計算されます。 ",
+  "retention.rule.attachButton.label.heading": "保持ルールを添付",
+  "retention.rule.attachButton.label.select": "添付するルールを選択",
+  "retention.rule.attachButton.label.placeholder": "ルール名を検索...",
+  "retention.rule.attachButton.label": "保持ルールを添付",
+  "retention.rule.attachButton.attached": "保持ルールが添付されました",
+  "retention.rule.attachButton.bulk.error": "保持ルールの添付中にエラーが発生しました: {0}",
+  "retention.rule.attachButton.bulk.label": "保持ルールを添付",
+  "retention.rule.attachButton.bulk.poll": "保持ルールを添付"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-ja.json:87]: same thing about new line

File: nuxeo-retention-web/i18n/messages-nl.json [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "De bewaartermijn wordt berekend op basis van de bewaartermijn die begint volgens het datumveld van een document. ",
+  "retention.rule.attachButton.label.heading": "Bevestig retentie-regel",
+  "retention.rule.attachButton.label.select": "Selecteer de regel om toe te voegen",
+  "retention.rule.attachButton.label.placeholder": "Zoek regel naam ...",
+  "retention.rule.attachButton.label": "Bevestig retentie-regel",
+  "retention.rule.attachButton.attached": "Retentie regel gekoppeld",
+  "retention.rule.attachButton.bulk.error": "Er is een fout opgetreden tijdens het koppelen van de eigenschapsregel: {0}",
+  "retention.rule.attachButton.bulk.label": "Bulkbevestiging bewaarregel",
+  "retention.rule.attachButton.bulk.poll": "Regel voor retentie koppelen"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-nl.json:87]: same thing about new line

File: nuxeo-retention-web/i18n/messages-pl.json [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "Okres retencji jest obliczany na podstawie rozpoczęcia czasu trwania retencji zgodnie z polem daty dokumentu",
+  "retention.rule.attachButton.label.heading": "Dołącz regułę retencji",
+  "retention.rule.attachButton.label.select": "Wybierz regułę do załączenia",
+  "retention.rule.attachButton.label.placeholder": "Szukaj nazwy reguły...",
+  "retention.rule.attachButton.label": "Dołącz regułę retencji",
+  "retention.rule.attachButton.attached": "Załączono regułę retencji",
+  "retention.rule.attachButton.bulk.error": "Wystąpił błąd podczas dołączania reguły retencji: {0}",
+  "retention.rule.attachButton.bulk.label": "Masowo dołącz regułę retencji",
+  "retention.rule.attachButton.bulk.poll": "Dołącz regułę retencji"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-pl.json:87]: same thing about new line

File: nuxeo-retention-web/i18n/messages-pt-PT.json [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "The retention period is calculated based on the retention duration starting according to a document's date field ",
+  "retention.rule.attachButton.label.heading": "Attach retention rule",
+  "retention.rule.attachButton.label.select": "Select the rule to be attached",
+  "retention.rule.attachButton.label.placeholder": "Search rule name ...",
+  "retention.rule.attachButton.label": "Attach retention rule",
+  "retention.rule.attachButton.attached": "Retention rule attached",
+  "retention.rule.attachButton.bulk.error": "An error occurred while attaching retention rule: {0}",
+  "retention.rule.attachButton.bulk.label": "Bulk attach retention rule",
+  "retention.rule.attachButton.bulk.poll": "Attaching retention rule"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-pt-PT.json:87]: same thing about new line

File: nuxeo-retention-web/i18n/messages.json [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "The retention period is calculated based on the retention duration starting according to a document's date field ",
+  "retention.rule.attachButton.label.heading": "Attach retention rule",
+  "retention.rule.attachButton.label.select": "Select the rule to be attached",
+  "retention.rule.attachButton.label.placeholder": "Search rule name ...",
+  "retention.rule.attachButton.label": "Attach retention rule",
+  "retention.rule.attachButton.attached": "Retention rule attached",
+  "retention.rule.attachButton.bulk.error": "An error occurred while attaching retention rule: {0}",
+  "retention.rule.attachButton.bulk.label": "Bulk attach retention rule",
+  "retention.rule.attachButton.bulk.poll": "Attaching retention rule"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages.json:87]: same thing about new line

File: nuxeo-retention-web/i18n/messages-zh-CN.json [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+  "retention.rule.label.startPolicy.metadataBased.description": "The retention period is calculated based on the retention duration starting according to a document's date field ",
+  "retention.rule.attachButton.label.heading": "Attach retention rule",
+  "retention.rule.attachButton.label.select": "Select the rule to be attached",
+  "retention.rule.attachButton.label.placeholder": "Search rule name ...",
+  "retention.rule.attachButton.label": "Attach retention rule",
+  "retention.rule.attachButton.attached": "Retention rule attached",
+  "retention.rule.attachButton.bulk.error": "An error occurred while attaching retention rule: {0}",
+  "retention.rule.attachButton.bulk.label": "Bulk attach retention rule",
+  "retention.rule.attachButton.bulk.poll": "Attaching retention rule"
+}
```
Comment (RSalem07) [inline nuxeo-retention-web/i18n/messages-zh-CN.json:87]: same thing about new line

File: nuxeo-retention-web/pom.xml [hunk 16]
```diff
@@ -0,0 +1,44 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.nuxeo.retention</groupId>
+    <artifactId>nuxeo-retention-parent</artifactId>
+    <version>11.0.0-SNAPSHOT</version>
```
Comment (RSalem07) [inline nuxeo-retention-web/pom.xml:6]: should be ` <version>11.4-SNAPSHOT</version>` otherwise it will not compile, tested on local

**Comment by RSalem07** [comment]
@nmpcunha I know it's not related to your ticket but wdyt if we rename `nuxeo-retention` to `nuxeo-retention-core` to be more consistent with what we did. I recall I had this discussion with Thomas about adding `-core` to be more consistent :)

**Comment by nmpcunha** [comment]
> @nmpcunha I know it's not related to your ticket but wdyt if we rename `nuxeo-retention` to `nuxeo-retention-core` to be more consistent with what we did. I recall I had this discussion with Thomas about adding `-core` to be more consistent :)

Guess we can move on that direction, but probably I would suggest for us to create a new ticket and do the same renaming on both addons: `retention` and `coldstorage`

**Comment by efge** [comment]
> @nmpcunha I know it's not related to your ticket but wdyt if we rename `nuxeo-retention` to `nuxeo-retention-core` to be more consistent with what we did. I recall I had this discussion with Thomas about adding `-core` to be more consistent :)

In the platform lots of modules are being merged together and I think `-core` goes away in many cases. But check with Thomas, yes.

**Comment by nmpcunha** [comment]
> > @nmpcunha I know it's not related to your ticket but wdyt if we rename `nuxeo-retention` to `nuxeo-retention-core` to be more consistent with what we did. I recall I had this discussion with Thomas about adding `-core` to be more consistent :)
> 
> In the platform lots of modules are being merged together and I think `-core` goes away in many cases. But check with Thomas, yes.

Thanks @efge!
Already clarified that with Thomas. We'll keep that as it is, for consistency purposes. 👍🏼

**Pull request merged**